salalib/mgraph.cpp

```diff
@@ -1630,16 +1630,17 @@ int MetaGraph::loadRT1(const std::vector<string>& fileset, Communicator *communi
    SuperSpacePixel::tail().m_region = QtRegion(map.getBottomLeft(), map.getTopRight());
 
    // for each category
-   for (size_t i = 0; i < map.size(); i++) {
-
-      SuperSpacePixel::tail().push_back(ShapeMap(map.key(i)));
-      SuperSpacePixel::tail().at(i).init(map.value(i).size(), map.getRegion() );
+   int i = -1;
+   for (auto val: map) {
+      i++;
+      SuperSpacePixel::tail().push_back(ShapeMap(val.first));

```

Comment:
Couldn't you initialise the shape map first and then push? Would simplify the loop, save one look-up and we could get rid of i.